### PR TITLE
Code Parser Refactoring

### PR DESCRIPTION
**Pull request by CDluznie**
Refactoring the  design of the parsers

### CODE REVIEW

File: tree_hugger/core/parser/php/php_parser.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import re
```
Comment (rcshubhadeep) [inline tree_hugger/core/parser/php/php_parser.py:1]: Skipping reviewing this one, as it is not very much done yet

File: tree_hugger/core/parser/php/queries.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-python_quaries:
+python_queries:
```
Comment (rcshubhadeep) [inline tree_hugger/core/parser/php/queries.yml:1]: Although most likely the top level key is going to go, but this key should have been `php_queries`

File: tree_hugger/core/parser/python/python_parser.py [hunk 3]
```diff
@@ -1,9 +1,12 @@
 import re
 from typing import List, Dict
 import logging
+from pathlib import Path
+import pkg_resources
```
Comment (rcshubhadeep) [inline tree_hugger/core/parser/python/python_parser.py:5]: Are we using this anymore?

File: example_queries/example_queries.yml [hunk 4]
```diff
@@ -1,4 +1,4 @@
-python_quaries:
+python_queries:
```
[the remark this answers is not in the document]
Comment (CDluznie) [reply example_queries/example_queries.yml:1]: Yes right

File: tree_hugger/core/code_parser.py [hunk 5]
```diff
@@ -25,10 +25,8 @@ class BaseParser(object):
     """
     Base parser exposes the common interface that we extend per language
     """
-    
-    DEFAULT_QUERY_FILE_PATH = str(Path("assets") / "queries" / "queries.yml")
 
-    def __init__(self, language: str, query_class_name: str, library_loc: str=None, query_file_path: str=None):
+    def __init__(self, language: str, query_class_name: str, query_file_path: str, library_loc: str=None):
```
[the remark this answers is not in the document]
Comment (CDluznie) [reply tree_hugger/core/code_parser.py:29]: `query_file_path` is not optional anymore and optional arguments must be at the end

File: tree_hugger/core/parser/python/python_parser.py (hunk 3 above)
Comment (CDluznie) [reply tree_hugger/core/parser/python/python_parser.py:5]: No, just forget to delete it, thanks

File: tree_hugger/core/parser/php/queries.yml (hunk 2 above)
Comment (CDluznie) [reply tree_hugger/core/parser/php/queries.yml:1]: I made the modification but forget to push, I can update if you want

File: MANIFEST.in [hunk 6]
```diff
@@ -1 +1,3 @@
-include tree_hugger/core/assets/queries/*.yml
\ No newline at end of file
+include tree_hugger/core/assets/queries/*.yml
```
Comment (rcshubhadeep) [inline MANIFEST.in:1]: Maybe we should get rid of this line @CDluznie

Comment (CDluznie) [reply MANIFEST.in:1]: Should we change by `.../parser/python/queries.yml` and same for others default parsers ?

Comment (rcshubhadeep) [reply MANIFEST.in:1]: No just delete the line. I have added all the needed lines bellow it. Please check

**Pull request merged**